### PR TITLE
[CSM] Hook ftruncate

### PR DESCRIPTION
**Pull request by Gui774ume**
## What does this PR do?

This PR hooks the ftruncate syscall.

## Motivation

This syscall was forgotten in the list of hooked syscalls. We currently support truncate but forgot ftruncate.

## Describe how to test/QA your changes

A functional test was added.

### CODE REVIEW

**Comment by lebauce** [comment]
@paulcacheux suggests to follow this one

**Review by paulcacheux — APPROVED** [review]

File: pkg/security/tests/open_test.go [hunk 1]
```diff
@@ -175,6 +175,51 @@ func TestOpen(t *testing.T) {
 		})
 	})
 
+	t.Run("ftruncate", func(t *testing.T) {
+		SkipIfNotAvailable(t)
+
+		defer os.Remove(testFile)
+
+		var f *os.File
+		test.WaitSignal(t, func() error {
```
Comment (spikat) [inline pkg/security/tests/open_test.go:184]: why do we need to call waitsignal here if we don't check it?

Comment (paulcacheux) [reply pkg/security/tests/open_test.go:184]: I'm not sure I think this is just an artifact of this being copied from the truncate test, will do a clean up of this in a follow up PR

**Review by spikat — APPROVED** [review]

File: pkg/security/tests/open_test.go (hunk 1 above)
Comment (spikat) [reply pkg/security/tests/open_test.go:184]: thanks!

**Review by lebauce — APPROVED** [review]

**Pull request merged**